### DISCUSSION TITLE
Lazy Loading of Accordion Panels

### DISCUSSION THREAD

**Discussion by ag288** [chakra-ui/chakra-ui#7704] [Q&A]
I have a number of accordion items. How can i ensure that the data in the accordion panel if fetched only when the accordion item is expanded?

**Comment by knovu** [comment]
I wonder if you could get the current accordion panel index, and simply conditional render or suspense?

**Comment by crhistianramirez** [comment]
@knovu your suggestion led me to the right path, it was actually a bit easier than your recommendation. You can access internal state via render prop. 

```html
<Accordion>
    <AccordionItem>
    {({isExpanded}) => (
        <>
        <h2>
            <AccordionButton _expanded={{bg: "tomato", color: "white"}}>
            <Box as="span" flex="1" textAlign="left">
                Click me to see a different style
            </Box>
            <AccordionIcon />
            </AccordionButton>
        </h2>
        <AccordionPanel pb={4}>
            {/* Don't render accordion panel unless viewing, enables lazy-loading for performance */}
            {isExpanded && (
            <p>
                Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et
                dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut
                aliquip ex ea commodo consequat.
            </p>
            )}
        </AccordionPanel>
        </>
    )}
    </AccordionItem>
</Accordion>
```

**Reply by knovu** [reply]
Awesome!  I just verified our frontend does the same, yea looks like our team lazy loads the content inside the panel.  So the "panel itself" is not lazy loaded, rather the content inside is. 

I guess you could patch in by making a wrapper, that does something similar to Chakra UI tabs prop: **isLazy**, that could enhance this as well.

**Reply by knovu** [reply]
I haven't verified this, but I wonder if you could wrap your stack of "AccordionItems" in "React.Suspense", and just lazy import directly into the content, so they all fallback to the same loader as well, in the Suspense prop = fallback.

I'm not 100% that would affect ref in any way, as maybe ref might get corrupted along the way to children.

**Reply by tylerpashigian** [reply]
This is a live saver, thank you!
